.changeset/blue-pans-suffer.md

```diff
@@ -0,0 +1,6 @@
+---
+"@asyncapi/generator": minor
+---
+- Moved the logic of **conditionalFile generation** to a new **conditionalGeneration module**, which now supports both file and folder generation in a unified way.
+- Introduced a new **conditionalGeneration** configuration that supports **subject** and **parameter** fields, enabling flexible and granular conditional generation.
+- The conditionalFile configuration is planned for deprecation in favor of the new conditionalGeneration setup. You can track the deprecation in the issue [#1553](https://github.com/asyncapi/generator/issues/1553).
```

Comment:
it is not planned for deprecation, this PR deprecates it, it is planned to be removed with future releases

Reply:
![image](https://github.com/user-attachments/assets/464d057f-23f1-455f-aacc-0feba37a95a1)
 updated please chekc thanks